src/Container/Container.php

```diff
@@ -165,4 +216,58 @@ public function __call($name, $arguments)
             throw new \BadMethodCallException("No application method named {$name}()");
         }
     }
+
+    /**
+     * Instantiate an object of named class, recursively resolving dependencies
+     *
+     * @param string $className Fully qualified class name
+     * @return mixed
+     * @throws \ReflectionException
+     */
+    protected function resolveInstance(string $className)
```

Comment:
Is this method intended to be used by the application's App instance? If not, it would be better to have it private instead to not expose any unnecessary complexity.

Reply:
Not really. It can be made private.